integreat_cms/cms/models/abstract_tree_node.py

```diff
@@ -282,7 +287,21 @@ def move(self, target, pos=None):
         # Moving a node can modify all other nodes via raw sql queries (which are not recognized by cachalot),
         # so we have to invalidate the whole model manually.
         invalidate_model(self.__class__)
-        super().move(target=target, pos=pos)
+        try:
+            with db_mutex(f"tree-page-{self.region}"):
+                super().move(target, pos)
+        except DBMutexError as e:
+            logger.error("Could not retrieve database mutex to save %s", self)
+            raise DBMutexError(
+                # pylint: disable=consider-using-f-string
+                _('Could not change position in tree of "{}".'.format(self))
+            ) from e
+        except DBMutexTimeoutError as e:
+            logger.error("Retrieving database mutex timed out while saving %s", self)
+            raise DBMutexTimeoutError(
+                # pylint: disable=consider-using-f-string
+                _('Could not change position in tree of "{}".'.format(self))
+            ) from e
```

Comment:
These exceptions should probably be caught in the view?
https://github.com/digitalfabrik/integreat-cms/blob/534f226f80618be350bb10c1e6d7c7448147db66/integreat_cms/cms/views/pages/page_actions.py#L494

Reply:
Ah yes, it should. Interestingly, I could not provoke the (timeout) exception yet. In all my tests (mostly with sleep) all processes waited for others to finish. And everything finished orderly.